src/utils.js

```diff
@@ -51,7 +51,12 @@ const buildReturnMessage = (utils, pass, property, received, expected) => () =>
   'Received:\n' +
   `  ${utils.printReceived(`${property}: ${received}`)}`;
 
-const matcherTest = (received, expected) => {
+const matcherTest = (received, expected, isNot) => {
+  // when negating, assert on existence of the style, rather than the value
+  if (isNot && expected === undefined) {
```

Comment:
Changed the strictness of the "falsey expected" condition. I believe the intention was to check for the existence of the second parameter to `toHaveStyleRule`, so "=== undefined" is better here. I can relax this, if there's concern around it being a breaking change.